OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,421 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Adds an overlay for weather animations based on dropping particles (snowfalkes, raindrops, sand grains)")]
+	public class WeatherOverlayInfo : ITraitInfo
+	{
+		[Desc("On-off switch for the weather overlay")]
+		public readonly bool WeatherEnabled = true;
```

Comment:
This seems redundant. If people don't want weather they can remove the trait. Also the Info classes don't work as on/off switches.
You can, however, have an `Enabled` property in the trait that you can set during the game, and even expose that to Lua to enable cool stuff to happen.